aiaccel/hpo/samplers/nelder_mead_sampler.py

```diff
@@ -0,0 +1,237 @@
+                search_space = {
+                    param_name: optuna.distributions.FloatDistribution(param_distribution[0], param_distribution[1])
+                    }
+                trans = _SearchSpaceTransform(search_space)
+                trans_params = self._rng.rng.uniform(trans.bounds[:, 0], trans.bounds[:, 1])
+                initial_param.append(trans.untransform(trans_params)[param_name])
+            self.NumofInitialCreateTrial += 1
+            yield np.array(initial_param), self.dimension + 1 - self.NumofInitialCreateTrial
+        temp = []
+        for _ in range(self.dimension + 1):
+            temp.append(self.vertex_queue.get())
+        self.vertices = np.array([vertex.coordinate for vertex in temp])
+        self.values = np.array([vertex.value for vertex in temp])
+
+    def order_by(self) -> None:
+        order = np.argsort(self.values)
+
+        self.vertices = self.vertices[order]
+        self.values = self.values[order]
+
+    def centroid(self) -> None:
+        self.storage = Storage()
+        self.order_by()
+        self.yc = self.vertices[:-1].mean(axis=0)
+
+    def reflect(self) -> Generator[np.ndarray[float, float], None, None]:
+        yr = self.yc + self.coef.r * (self.yc - self.vertices[-1])
+        self.storage.r = yr
+        yield from self.func(yr)
+
+    def expand(self) -> Generator[np.ndarray[float, float], None, None]:
+        ye = self.yc + self.coef.e * (self.yc - self.vertices[-1])
+        self.storage.e = ye
+        yield from self.func(ye)
+
+    def inside_contract(self) -> Generator[np.ndarray[float, float], None, None]:
+        yic = self.yc + self.coef.ic * (self.yc - self.vertices[-1])
+        self.storage.ic = yic
+        yield from self.func(yic)
+
+    def outside_contract(self) -> Generator[np.ndarray[float, float], None, None]:
+        yoc = self.yc + self.coef.oc * (self.yc - self.vertices[-1])
+        self.storage.oc = yoc
+        yield from self.func(yoc)
+
+    def shrink(self) -> Generator[np.ndarray[float, float], None, None]:
+        for i in range(1, len(self.vertices)):
+            yield self.vertices[0] + self.coef.s * (self.vertices[i] - self.vertices[0]), 0
+        for i in range(1, len(self.vertices)):
+            vertex = self.vertex_queue.get()
+            self.vertices[i] = vertex.coordinate
+            self.values[i] = vertex.value
+
+    def func(self, y: np.ndarray[float, float]) -> Generator[np.ndarray[float, float], None, None]:
+        if self.is_within_range(y):
+            yield y, 0
+        else:
+            self.vertex_queue.put(Vertex(y, np.inf))
+
+    def is_within_range(self, coordinates: np.ndarray[float, float]) -> bool:
+        return all(not (co < ss[0] or ss[1] < co) for ss, co in zip(self._search_space.values(), coordinates))
+
+    def search(self) -> Generator[np.ndarray[float, float], None, None]:
+        yield from self.initial()
+        while True:
+
+            self.centroid()
```

Comment:
副作用のみの関数だと，何をしているのかぱっと分からないので，
```suggestion
            yc = self.centroid(self.xxx)
```
のように，入出力で作用が分かるように書いて頂いても良いでしょうか．

Reply:
> 数理的なアルゴリズムの実装において，1回しか使われない・使う予定のないメソッドは作成する必要がないように思います．
こちらの改修の都合で self.centroid 関数も削除されました。